sympy/solvers/solveset.py

```diff
@@ -539,23 +539,7 @@ def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        return Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
```

Comment:
Even though if `y` is substituted then also [this line](https://github.com/Shekharrajak/sympy/blob/master/sympy/solvers/tests/test_solveset.py#L749) not always able to give simple form.So there are some chance that solution will be [`FiniteSet`](https://github.com/sympy/sympy/pull/10482#discussion-diff-51149813L552).
Hence many times it end up [here](https://github.com/sympy/sympy/pull/10482#discussion-diff-51149813L557) with `conditionSet`, that we don't want.